### PR TITLE
fix(engine): Generate compact subgraph queries

### PR DESCRIPTION
**Pull request by Finistere**
Helps when subgraph log the queries to stdout at least.
Also adding a feature flag `pprof` for the gateway to have quickly a profile, it's quite convenient.

### CODE REVIEW

File: engine/crates/integration-tests/tests/federation/auth/authenticated.rs [hunk 1]
```diff
@@ -101,22 +101,22 @@ fn not_authenticated() {
 
         // We shouldn't have requested the field.
         let requests = engine.drain_graphql_requests_sent_to::<SecureSchema>();
-        insta::assert_json_snapshot!(requests, @r###"
+        insta::assert_json_snapshot!(requests, @r#"
         [
           {
-            "query": "query {\n  check {\n    anonymous\n  }\n}\n",
+            "query": "query{ check{ anonymous}}",
```
Comment (pimeys) [inline engine/crates/integration-tests/tests/federation/auth/authenticated.rs:107]: Do we need the whitespace after `{`'s?

**Review by pimeys — APPROVED** [review]

File: engine/crates/integration-tests/tests/federation/auth/authenticated.rs (hunk 1 above)
Comment (Finistere) [reply engine/crates/integration-tests/tests/federation/auth/authenticated.rs:107]: It's just to handle correctly multiple fields. one space won't matter, the actual problematic part were newlines

File: gateway/crates/gateway-binary/src/main.rs [hunk 2]
```diff
@@ -19,6 +19,13 @@ mod telemetry;
 const THREAD_NAME: &str = "grafbase-gateway";
 
 fn main() -> anyhow::Result<()> {
+    #[cfg(feature = "pprof")]
+    let guard = pprof::ProfilerGuardBuilder::default()
+        .frequency(5000)
+        .blocklist(&["libc", "libgcc", "pthread", "vdso"])
+        .build()
+        .unwrap();
```
Comment (obmarg) [inline gateway/crates/gateway-binary/src/main.rs:27]: Can these pprof bits go in functions (possibly with the `cfg inside the functions?)

Minor complaint, just seems irrelevant most of the time so would be good to not clutter up main so much

File: engine/crates/engine-v2/src/sources/graphql/request/prepare.rs [hunk 3]
```diff
@@ -97,7 +89,7 @@ impl PreparedFederationEntityOperation {
         );
         query.push_str("query");
         query.push('(');
-        write!(query, "${entities_variable_name}: [_Any!]!")?;
```
Comment (obmarg) [inline engine/crates/engine-v2/src/sources/graphql/request/prepare.rs:100]: Nitpicking a bit, but surely a few spaces for readability would be ok...

**Review by obmarg — APPROVED** [review]

File: engine/crates/engine-v2/src/sources/graphql/request/prepare.rs (hunk 3 above)
Comment (Finistere) [reply engine/crates/engine-v2/src/sources/graphql/request/prepare.rs:100]: fair enough

File: gateway/crates/gateway-binary/src/main.rs (hunk 2 above)
Comment (Finistere) [reply gateway/crates/gateway-binary/src/main.rs]: separated it from the rest, not great, but better than before at least
